R/dockerfile.R

```diff
@@ -27,38 +19,130 @@
 #' dockerfile()
 #'
 #' @import futile.logger
-dockerfile <- function(from = sessionInfo(), to = paste0(getwd(), "/", "Dockerfile")) {
-  flog.debug("Creating a new Dockerfile from %s to %s", from, to)
+dockerfile <- function(from = utils::sessionInfo(),env = NULL, maintainer = NULL, r_version = paste(R.Version()$major, R.Version()$minor, sep="."), image = NULL, context = NA_character_) {
+  flog.debug("Creating a new Dockerfile from %s", from)
   .dockerfile <- NA
   .originalFrom <- class(from)
-
+  
+  #Instructions that create a basic dockerfile (from could be NULL)
+  
+  if(is.null(image)) {
+    #check if dockerized R version is available (maybe check other repositories too?)
+    tags <- tagsfromRemoteImage("rocker/r-ver")
+      if(r_version %in% tags){
+        image=From("rocker/r-ver",tag = r_version)
+      }else{
+        stop("No docker image found for the given R version. ", 
+            "Please either specify a custom Docker image or \n", 
+             "  use one of the following supported version tags (maybe check the internet connection). \n\t", 
+             paste(tags, collapse = " "))
+      }
+  } 
+  #parse From-object from string if necessary
+  if (is.character(image)) {
+    image = parseFrom(image)
+  }
+  
+  instructions=list()
+  
+  cmd = Cmd("R") ###default CMD may be overwritten e.g. from dockerfileFromSession
+  
+  .dockerfile = new("Dockerfile", instructions=instructions, maintainer=maintainer, image=image, context=context, cmd=cmd)
+  
+  
   if(inherits(x = from, "sessionInfo")) {
-    .dockerfile <- dockerfileFromSession(session = from, to = to)
+    .dockerfile <- dockerfileFromSession(session = from, .dockerfile = .dockerfile)
   } else if (inherits(x = from, "file")) {
-    .dockerfile <- dockerfileFromFile(file = from, to = to)
+    .dockerfile <- dockerfileFromFile(file = from, .dockerfile = .dockerfile)
   } else if(inherits(x = from, "character") && dir.exists(from)) {
-    .dockerfile <- dockerfileFromWorkspace(path = from, to)
+    .dockerfile <- dockerfileFromWorkspace(path = from, .dockerfile = .dockerfile)
     .originalFrom <- from
-  } else {
+  }else if(is.null(from)) {
+      #Creates a basic dockerfile without the 'from'-argument 
+  }else {
     stop("Unsupported 'from': ", class(from), from)
   }
-
-  flog.info("Created Dockerfile at %s based on %s", to, .originalFrom)
-  message("Created Dockerfile at", to, "based on", .originalFrom)
+  
+  flog.info("Created Dockerfile-Object based on %s", .originalFrom)
+  message("Created Dockerfile-Object based on ", .originalFrom, ".")
   return(.dockerfile)
 }
 
+#' Format a dockerfile object to a series of instruction
+#'
+#' @param x An object of class Dockerfile
+#' @param ... Arguments to be passed down to format.default
+#'
+#' @return The Content of the dockerfile represented by the dockerfile object, by default formated as a List of strings where each string represent a new line
```

Comment:
"the content", "list of strings" (no capitals for regular words), but "Dockerfile" (because it is a name!)